app/src/main/java/com/manage/ak/moneyreport/MainActivity.java

```diff
@@ -0,0 +1,441 @@
+package com.manage.ak.moneyreport;
+
+import android.app.Dialog;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.content.SharedPreferences;
+import android.database.Cursor;
+import android.net.Uri;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.CardView;
+import android.support.v7.widget.Toolbar;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.Window;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MainActivity extends AppCompatActivity implements View.OnClickListener {
+
+    private Context context = this;
+
+    // A list to store bank list and cash sms
+    private List<Sms> bankList = new ArrayList<>();
+    private List<Sms> cashList = new ArrayList<>();
+
+    // Main balance in the bank
+    // In many places the balance is needed in String form.
+    private String BALANCE = "0.0";
+    // Total cash spent from cash in hand
+    private String CASHSPENT = "0.0";
+
+    // variables used to store and calculate balance for the cash transactions
+    private String amt, type, balance;
+
+    // bank balance and date TextView inside the bank card
+    private TextView bankBalance;
+    private TextView estimateDate;
+
+    // spent amount and cash in hand inside the cash card
+    private TextView spentAmount;
+    private TextView cashBalance;
+
+    // BroadcastReceiver listening to the incoming messages
+    private static final String SMS_RECEIVED = "android.provider.Telephony.SMS_RECEIVED";
+    IntentFilter intentFilter = new IntentFilter(SMS_RECEIVED);
+    private BroadcastReceiver sms_receiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            if (intent.getAction().equals(SMS_RECEIVED)) {
+                readMessages();
```

Comment:
I am a bit confused in this. When a new messages comes you directly call the readMessages where you read messages from your DB. But it needs to get first inserted into the DB to get created.